### PR TITLE
Add phpstan

### PR DESCRIPTION
**Pull request by alexmanno**

### CODE REVIEW

File: src/AutoloadPathProvider.php [hunk 1]
```diff
@@ -45,7 +45,9 @@ public function getPaths(): array
             throw new \RuntimeException('Unable to find composer.json');
         }
 
-        $composer = \json_decode(\file_get_contents($this->composerPath), true);
+        /** @var string $composerContent */
```
Comment (thomasvargiu) [inline src/AutoloadPathProvider.php]: We should handle failure on `file_get_contents()` and `json_decode()` calls

File: src/Installer/Installer.php [hunk 2]
```diff
@@ -68,7 +68,7 @@ public function __construct(
         // Get composer.json location
         $composerFile = $composerPath ?: Factory::getComposerFile();
         // Calculate project root from composer.json, if necessary
-        $this->projectRoot = $projectRoot ?: \realpath(\dirname($composerFile));
+        $this->projectRoot = (string) ($projectRoot ?: \realpath(\dirname($composerFile)));
```
Comment (thomasvargiu) [inline src/Installer/Installer.php]: We shouldn't cast, we should handle `realpath` failure in some way.

We can start checking for `\dirname($this->composerPath)` existence.

File: src/Installer/Installer.php [hunk 3]
```diff
@@ -157,9 +157,13 @@ private function parseComposerDefinition(Composer $composer, $composerFile): voi
         $this->composerJson = new JsonFile($composerFile);
         $this->composerDefinition = $this->composerJson->read();
         // Get root package
-        $this->rootPackage = $composer->getPackage();
+        /** @var BasePackage $rootPackage */
```
Comment (thomasvargiu) [inline src/Installer/Installer.php]: `$composer->getPackage` returns a `PackageInterface`, so we can't assume that `$rootPackage` it's an instance of it.

Btw, ` $rootPackage` property is private and seems not used, so we can remove it without a BC Break.

File: .travis.yml [hunk 4]
```diff
@@ -50,6 +51,7 @@ install:
 script:
   - if [[ $EXECUTE_TEST_SCRUTINIZER == 'true' ]]; then ./vendor/bin/phpunit --coverage-clover coverage.clover ; fi
   - if [[ $EXECUTE_TEST_SCRUTINIZER != 'true' ]]; then ./vendor/bin/phpunit ; fi
+  - if [[ $EXECUTE_TEST_PHPSTAN != 'true' ]]; then ./vendor/bin/phpstan analyse -c phpstan.neon -l7 . ; fi
```
Comment (thomasvargiu) [inline .travis.yml]: `-c phpstan.neon` is not necessary. We should add `--no-progress` flag too

Comment (alexmanno) [reply .travis.yml]: `-c phpstan.neon` is necessary in `phpstan:^0.9`

File: src/AutoloadPathProvider.php [hunk 5]
```diff
@@ -31,7 +31,17 @@ class AutoloadPathProvider
     public function __construct(?string $composerFile = null, ?string $projectRoot = null, bool $dev = true)
     {
         $this->composerPath = $composerFile ?: \trim(\getenv('COMPOSER') ?: '') ?: './composer.json';
```
Comment (thomasvargiu) [inline src/AutoloadPathProvider.php]: What do you think to simplify with something like that?

```php
$this->projectRoot = $projectRoot ?: \realpath(\dirname($this->composerPath));

if (false === $this->projectRoot) {
    throw new \RuntimeException('Unable to get project root.');
}
```

File: src/Installer/Installer.php [hunk 6]
```diff
@@ -68,7 +61,15 @@ public function __construct(
         // Get composer.json location
         $composerFile = $composerPath ?: Factory::getComposerFile();
         // Calculate project root from composer.json, if necessary
-        $this->projectRoot = (string) ($projectRoot ?: \realpath(\dirname($composerFile)));
```
Comment (thomasvargiu) [inline src/Installer/Installer.php]: What do you think to simplify with something like that?

```php
$this->projectRoot = $projectRoot ?: \realpath(\dirname($this->composerPath));

if (false === $this->projectRoot) {
    throw new \RuntimeException('Unable to get project root.');
}
```

**Pull request merged**